### PR TITLE
Add recycling vendor

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Implemented that doc:
https://docs.google.com/document/d/1BHpzbeIeibqx00sTaVmgCnVk_VgOiZ-CwxKrFQgeyv0

image

image

image

image

image

TL;DR: Guild get machine in the lobby that buys materials from people.

## Why It's Good For The Game

Content cool.

## Changelog
:cl:
add: recycling vendor
/:cl:

### CODE REVIEW

**Review by TheShown911 — COMMENTED** [review]
Guild buying materials for base price? You can't have guild without ripping people off.

**Comment by SirRichardFrancis** [comment]
> Guild buying materials for base price? You can't have guild without ripping people off.

Then there is no incentive to sell materials, prices are laughably low as is.

**Review by TheShown911 — APPROVED** [review]

File: code/game/machinery/recycle_vendor.dm [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	if(stat & BROKEN)
+		icon_state = "recycle_broken"
+		return
+
+	icon_state = "recycle"
+
+	if(stat & NOPOWER || !anchored)
+		return
+
+	overlays += sales_paused || !materials_allowed.len	? "recycle_screen_red"			: "recucle_screen_green"
```
Comment (hyperioo) [inline code/game/machinery/recycle_vendor.dm:64]: I was going to say you got a typo but you actually also named the overlay "recucle" so I guess it checks out.

File: code/game/machinery/recycle_vendor.dm [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+			materials_stored.Add(i)
+			materials_stored[i] = stored_item_materials[i]
+
+	playsound(loc, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 50, 1)
+	spawn_money(stored_item_value, loc)
+	flick("recycle_vend", src)
+	update_icon()
+
+	if(stored_item_value < 50 && prob(5))
+		speak("Ты бы еще консервных банок насобирал!")
```
Comment (hyperioo) [inline code/game/machinery/recycle_vendor.dm:202]: Bruh

File: code/game/machinery/recycle_vendor.dm (hunk 1 above)
Comment (SirRichardFrancis) [reply code/game/machinery/recycle_vendor.dm:64]: Yeah, I fat fingered it.

**Pull request merged**